### PR TITLE
Trigger default action when open file initial state is set

### LINKED ISSUE #41467: [Bug]: shared file link opens folder, not file
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

opening a shared internal link that is pointing to a file, i.e. https://mynextcloud.url/f/1234567 in a browser opens the folder, not the file itself.

This is seen as a regression.

### Steps to reproduce

1. copy/share/receive an internal link to a file eg. https://mynextcloud.url/f/1234567
2. click on / browse to said link in browser
3. the folder that contains the file is indeed opened, but the file itself is not opened in eg. Files editor for .md

### Expected behavior

expecting the file that was linked to be opened in Files editor for .md, etc.

### Installation method

None

### Nextcloud Server version

28.0.0 beta 3

### Operating system

Other

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

Updated from a MINOR version (ex. 22.1 to 22.2)

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

Nextcloud Server version 28.0.0 beta 2
Firefox & Chromium 
openSUSE Tumbleweed

### PR DESCRIPTION
**Pull request by artonge**
Reimplementation of https://github.com/nextcloud/server/pull/27102 for F2V
Fix: https://github.com/nextcloud/server/issues/41467

### CODE REVIEW

File: apps/files/src/components/FilesListVirtual.vue [hunk 1]
```diff
@@ -73,17 +73,18 @@
 </template>
 
 <script lang="ts">
-import type { Node as NcNode } from '@nextcloud/files'
+import { getFileActions, Node as NcNode, getFileListHeaders, Folder, View, Permission } from '@nextcloud/files'
```
Comment (susnux) [inline apps/files/src/components/FilesListVirtual.vue:76]: type only imports should be kept

```suggestion
import type { Node as NcNode } from '@nextcloud/files'
```

File: apps/files/src/components/FilesListVirtual.vue [hunk 2]
```diff
@@ -73,17 +73,18 @@
...
 import type { PropType } from 'vue'
 import type { UserConfig } from '../types.ts'
 
 import { Fragment } from 'vue-frag'
-import { getFileListHeaders, Folder, View, Permission } from '@nextcloud/files'
```
Comment (susnux) [inline apps/files/src/components/FilesListVirtual.vue:81]: see above

File: apps/files/src/components/FilesListVirtual.vue [hunk 3]
```diff
@@ -73,17 +73,18 @@
...
 import { showError } from '@nextcloud/dialogs'
+import { loadState } from '@nextcloud/initial-state'
 import { translate as t, translatePlural as n } from '@nextcloud/l10n'
 import Vue from 'vue'
 
 import { action as sidebarAction } from '../actions/sidebarAction.ts'
+import { action as openInFilesAction } from '../actions/openInFilesAction.ts'
```
Comment (susnux) [inline apps/files/src/components/FilesListVirtual.vue:87]: never used?

File: apps/files/src/components/FilesListVirtual.vue [hunk 4]
```diff
@@ -217,6 +218,8 @@ export default Vue.extend({
 
 		this.scrollToFile(this.fileId)
 		this.openSidebarForFile(this.fileId)
+		this.handleOpenFile()
```
Comment (susnux) [inline apps/files/src/components/FilesListVirtual.vue:221]: Probably also add this to the `watch: fileId` above, no?

**Review by susnux — COMMENTED** [review]
Nice! But got some comments for you

File: apps/files/src/components/FilesListVirtual.vue (hunk 4 above)
Comment (artonge) [reply apps/files/src/components/FilesListVirtual.vue:221]: No, as this is base on the initial state.

File: apps/files/src/components/FilesListVirtual.vue (hunk 1 above)
Comment (artonge) [reply apps/files/src/components/FilesListVirtual.vue:76]: Stupid auto format... thanks

**Review by Pytal — APPROVED** [review]

**Review by juliusknorr — APPROVED** [review]
Tested and works 👍

**Pull request merged**